lib/read_preference.js

```diff
@@ -16,19 +16,8 @@ const ReadPreference = function(mode, tags, options) {
     throw new TypeError(`Invalid read preference mode ${mode}`);
   }
 
-  // TODO(major): tags MUST be an array of tagsets
-  if (tags && !Array.isArray(tags)) {
-    console.warn(
-      'ReadPreference tags must be an array, this will change in the next major version'
-    );
-
-    if (typeof tags.maxStalenessSeconds !== 'undefined') {
-      // this is likely an options object
-      options = tags;
-      tags = undefined;
-    } else {
-      tags = [tags];
-    }
+  if (!Array.isArray(tags)) {
```

Comment:
Hoooold up! `new ReadPreference('primary')` is a valid read preference. `tags` _are_ optional, so we can't just throw an error if they aren't an array.

Reply:
Yeah that makes sense - I think we were thrown off by the jsdoc showing it as required. I'll revert this change and update the jsdoc to show tags as optional.